File: Prueba y error/ejercicios de semana santa java/one java/Library/HorasExtras.java
package Library;

public class HorasExtras {
    private double valorHora;
    private int diurnas;
    private int nocturnas;
    private int festivas;

    public HorasExtras(double valorHora, int diurnas, int nocturnas, int festivas) {
        this.valorHora = valorHora;
        this.diurnas = diurnas;
        this.nocturnas = nocturnas;
        this.festivas = festivas;
    }

    public double calcularTotal() {
        double total = 0;
        total += diurnas * valorHora * 1.25;
        total += nocturnas * valorHora * 1.75;
        total += festivas * valorHora * 2.00;
        return total;
    }
}

File: Prueba y error/ejercicios de semana santa php/One php/Library/HorasExtras.php
<?php
namespace Library;

class HorasExtras {
    private float $valorHora;
    private int $horasDiurnas;
    private int $horasNocturnas;
    private int $horasFestivas;

    public function __construct(float $valorHora, int $horasDiurnas, int $horasNocturnas, int $horasFestivas) {
        $this->valorHora = $valorHora;
        $this->horasDiurnas = $horasDiurnas;
        $this->horasNocturnas = $horasNocturnas;
        $this->horasFestivas = $horasFestivas;
    }

    public function calcularTotal(): float {
        $totalHorasExtras = ($this->horasDiurnas * $this->valorHora) +
                            ($this->horasNocturnas * $this->valorHora * 1.35) +
                            ($this->horasFestivas * $this->valorHora * 1.75);
        return $totalHorasExtras;
    }
}